src/main/java/co/cask/hydrator/plugin/batch/DynamoDBBatchSource.java

```diff
@@ -0,0 +1,243 @@
+    @Macro
+    private String placeholderType;
+
+    @Description("Read Throughput for AWS DynamoDB table to connect to, in double. Default is 1. (Macro Enabled)")
+    @Nullable
+    @Macro
+    private String readThroughput;
+
+    @Description("Read Throughput Percentage for AWS DynamoDB table to connect to. Default is 0.5. (Macro Enabled)")
+    @Nullable
+    @Macro
+    private String readThroughputPercentage;
+
+    public DynamoDBConfig(String referenceName, String accessKey, String secretAccessKey, @Nullable String regionId,
+                          @Nullable String endpointUrl, String tableName, String query, @Nullable String filterQuery,
+                          @Nullable String nameMappings, String valueMappings, String placeholderType,
+                          @Nullable String readThroughput, @Nullable String readThroughputPercentage) {
+      super(referenceName);
+      this.accessKey = accessKey;
+      this.secretAccessKey = secretAccessKey;
+      this.regionId = regionId;
+      this.endpointUrl = endpointUrl;
+      this.tableName = tableName;
+      this.query = query;
+      this.filterQuery = filterQuery;
+      this.nameMappings = nameMappings;
+      this.valueMappings = valueMappings;
+      this.placeholderType = placeholderType;
+      this.readThroughput = readThroughput;
+      this.readThroughputPercentage = readThroughputPercentage;
+    }
+
+    /**
+     * Validates whether the table name follows the DynamoDB naming rules and conventions or not.
+     */
+    private void validateTableName() {
+      int tableNameLength = tableName.length();
+      if (tableNameLength < 3 || tableNameLength > 255) {
+        throw new IllegalArgumentException(
+          String.format("Table name '%s' does not follow the DynamoDB naming rules. Table name must be between 3 and " +
+                          "255 characters long.", tableName));
+
+      }
+
+      String pattern = "^[a-zA-Z0-9_.-]+$";
+      Pattern patternObj = Pattern.compile(pattern);
+
+      if (!Strings.isNullOrEmpty(tableName)) {
+        Matcher matcher = patternObj.matcher(tableName);
+        if (!matcher.find()) {
+          throw new IllegalArgumentException(
+            String.format("Table name '%s' does not follow the DynamoDB naming rules. Table names can contain only " +
+                            "the following characters: 'a-z, A-Z, 0-9, underscore(_), dot(.) and dash(-)'.",
+                          tableName));
+        }
+      }
+    }
+
+    /**
+     * Validates whether the endpointUrl is in proper format or not.
+     */
+    private void validateEndpointURL() {
+      if (!endpointUrl.startsWith("http://") || !endpointUrl.startsWith("https://")) {
+        throw new IllegalArgumentException(
+          String.format("Invalid endpoint URL '%s'. Please make sure that the URL is in the expected format i.e. " +
+                          "http://<host>:<port> or https://<host>:<port>.", endpointUrl));
+      }
```

Comment:
Removed the validation for endpointUrl.
User will enter the valid endpoint URL for establishing the connection. Also, updated the description for endpointUrl parameter.